src/main/java/net/snowflake/client/core/SessionUtil.java

```diff
@@ -259,12 +265,19 @@ static SFLoginOutput openSession(
       }
     }
 
-    boolean isClientStoreTemporaryCredential =
-        asBoolean(loginInput.getSessionParameters().get(CLIENT_STORE_TEMPORARY_CREDENTIAL));
-    if (isClientStoreTemporaryCredential) {
-      CredentialManager.getInstance().fillCachedIdToken(loginInput);
+    if (authenticator.equals(ClientAuthnDTO.AuthenticatorType.USERNAME_PASSWORD_MFA)) {
+      if (Constants.getOS() == Constants.OS.MAC || Constants.getOS() == Constants.OS.WINDOWS) {
+        loginInput.getSessionParameters().put(CLIENT_REQUEST_MFA_TOKEN, true);
+      } else {
+        if (!loginInput.getSessionParameters().containsKey(CLIENT_REQUEST_MFA_TOKEN)) {
+          // for testing purpose
+          loginInput.getSessionParameters().put(CLIENT_REQUEST_MFA_TOKEN, false);
+        }
```

Comment:
Linux's local file cache is supported internally at present. This feature will be announced only for Mac & Windows this time.